pkg/util/validation/validate.go

```diff
@@ -339,6 +340,13 @@ func ValidateMetadata(validateMetrics *ValidateMetrics, cfg *Limits, userID stri
 
 func ValidateNativeHistogram(validateMetrics *ValidateMetrics, limits *Limits, userID string, ls []cortexpb.LabelAdapter, histogramSample cortexpb.Histogram) (cortexpb.Histogram, error) {
 
+	// sample size validation for native histogram
+	nhSampleSize := histogramSample.Size()
```

Comment:
Nit: we can only calculate the sample size if the limit is > 0.

Reply:
Thanks. Updated.